libraries/Microsoft.Bot.Builder.Core/Adapters/ConsoleAdapter.cs

```diff
@@ -21,29 +22,10 @@ public new ConsoleAdapter Use(IMiddleware middleware)
             return this;
         }
 
-        public async Task ProcessActivity(Func<ITurnContext, Task> callback = null)
+        public override async Task ProcessActivity(Activity activity, Func<ITurnContext, Task> callback = null, CancellationTokenSource cancelToken = null)
```

Comment:
So I don't think you want to take a `CancellationToken*Source*` here, you want to take just a `CancellationToken`. `CancellationTokenSource` is for the person who is in control of the actual cancellation itself which is usually upstream in the "app", where a method like this takes a `CancellationToken` to know when cancellation has occurred so that it can stop working sooner than it might have. Make sense?

FWIW, there's a side-effect to this change that you'll run into given you've made it an optional parameter and that is that you would need to do `CancellationToken cancellationToken = default(CancellationToken)` instead of `= null` because it's  a value type.

(NOTE: I'll only put this comment here once, but obviously you need to apply it to the base class and that will have ripples through all the overrides)